### PR TITLE
yolov8 seg demo

### LINKED ISSUE #131: Demonstrate NN background removal (blur)
TODO

### PR DESCRIPTION
**Pull request by denisvmedyantsev**

### CODE REVIEW

File: samples/yolov8_seg/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# YOLOv8 Instance Segmentation
+
+A simple pipeline using a [YOLOv8 instance segmentation model](https://docs.ultralytics.com/tasks/segment/) to identify the people in a frame and to segment them from the rest of the frame.
+
+We created an ONNX version of the YOLOv8m-seg model using a script from the original repository (see [Export section](https://docs.ultralytics.com/tasks/segment/#export)). To process the model output, we wrote [convertor](module/converter.py). The segmentation model is a complex model in Savant terms. The model produces objects defined by bounding boxes and corresponding masks. To render the bounding boxes and masks we used cv2.cuda.GpuMat, the implementation code is in the [overlay](module/overlay.py).
```
Comment (abramov-oleg) [inline samples/yolov8_seg/README.md]: ```[convertor](module/converter.py)```

Convertor / converter spelling inconsistency.

I suggest `converter`

File: samples/yolov8_seg/module/converter.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        tensors[:, [2, 4]] *= ratio_x
+        tensors[:, [3, 5]] *= ratio_y
+        tensors[:, 2] += roi_left
+        tensors[:, 3] += roi_top
+
+        # scale masks (transpose to use cv2.resize)
+        masks = masks.transpose((1, 2, 0))
+        masks = cv2.resize(
+            masks,
+            (int(ratio_x * model.input.width), int(ratio_y * model.input.height)),
```
Comment (abramov-oleg) [inline samples/yolov8_seg/module/converter.py:80]: If resize ratios were calculated then it is possible to use them directly in the `cv2.resize()` call

Something along the lines of

```
cv2.resize(
        masks,
        dsize=None,
        fx=ratio_x,
        fy=ratio_y,
        cv2.INTER_LINEAR,
```

File: samples/yolov8_seg/README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# YOLOv8 Instance Segmentation
```
Comment (abramov-oleg) [inline samples/yolov8_seg/README.md:1]: Our samples usually have a .webp preview.

Can we add one for this demo too?

Comment (abramov-oleg) [inline samples/yolov8_seg/README.md:1]: Let's add a link to this demo from `samples/README.md`

File: samples/yolov8_seg/module/module.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+          # COCO classes {0: 'person', 1: 'bicycle', 2: 'car', 3: 'motorcycle', 4: 'airplane', 5: 'bus', 6: 'train', 7: 'truck', 8: 'boat', 9: 'traffic light', 10: 'fire hydrant', 11: 'stop sign', 12: 'parking meter', 13: 'bench', 14: 'bird', 15: 'cat', 16: 'dog', 17: 'horse', 18: 'sheep', 19: 'cow', 20: 'elephant', 21: 'bear', 22: 'zebra', 23: 'giraffe', 24: 'backpack', 25: 'umbrella', 26: 'handbag', 27: 'tie', 28: 'suitcase', 29: 'frisbee', 30: 'skis', 31: 'snowboard', 32: 'sports ball', 33: 'kite', 34: 'baseball bat', 35: 'baseball glove', 36: 'skateboard', 37: 'surfboard', 38: 'tennis racket', 39: 'bottle', 40: 'wine glass', 41: 'cup', 42: 'fork', 43: 'knife', 44: 'spoon', 45: 'bowl', 46: 'banana', 47: 'apple', 48: 'sandwich', 49: 'orange', 50: 'broccoli', 51: 'carrot', 52: 'hot dog', 53: 'pizza', 54: 'donut', 55: 'cake', 56: 'chair', 57: 'couch', 58: 'potted plant', 59: 'bed', 60: 'dining table', 61: 'toilet', 62: 'tv', 63: 'laptop', 64: 'mouse', 65: 'remote', 66: 'keyboard', 67: 'cell phone', 68: 'microwave', 69: 'oven', 70: 'toaster', 71: 'sink', 72: 'refrigerator', 73: 'book', 74: 'clock', 75: 'vase', 76: 'scissors', 77: 'teddy bear', 78: 'hair drier', 79: 'toothbrush'}
```
Comment (abramov-oleg) [inline samples/yolov8_seg/module/module.yml]: Is the full coco classes list in a single line necessary here?

Seems too verbose.

**Review by abramov-oleg — CHANGES_REQUESTED** [review]

File: samples/yolov8_seg/README.md (hunk 3 above)
Comment (denisvmedyantsev) [reply samples/yolov8_seg/README.md:1]: Added

Comment (denisvmedyantsev) [reply samples/yolov8_seg/README.md:1]: Done

File: samples/yolov8_seg/README.md (hunk 1 above)
Comment (denisvmedyantsev) [reply samples/yolov8_seg/README.md]: Replaced with `converter`.

File: samples/yolov8_seg/module/converter.py (hunk 2 above)
Comment (denisvmedyantsev) [reply samples/yolov8_seg/module/converter.py:80]: Model input 640x640, mask output 160x160, I can't use `ratio_x` directly.

File: samples/yolov8_seg/module/module.yml (hunk 4 above)
Comment (denisvmedyantsev) [reply samples/yolov8_seg/module/module.yml]: Removed.

**Review by abramov-oleg — APPROVED** [review]

**Pull request merged**